src/Kernel/Services/ChargeService.php

```diff
@@ -209,4 +236,94 @@ public function cancel(Charge $charge, $amount = 0)
 
         return new ServiceResponse(false, $resultApi);
     }
-}
\ No newline at end of file
+
+    /**
+     * @param Charge[] $listCharge
+     * @return Charge[]|array
+     */
+    public function getNotFailedOrCanceledCharges(array $listCharge)
+    {
+        $existStatusFailed = array_filter(
+            $listCharge,
+            function (Charge $charge) {
+                return (
+                    ($charge->getStatus()->getStatus() != 'failed') &&
+                    ($charge->getStatus()->getStatus() != 'canceled')
+                );
+            }
+        );
+
+        return $existStatusFailed;
+    }
+
+    public function prepareHistoryComment(ChargeInterface $charge)
+    {
+        $i18n = new LocalizationService();
+        $moneyService = new MoneyService();
+
+        if (
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid using static access to class '\Mundipagg\Core\Kernel\ValueObjects\ChargeStatus' in method 'prepareHistoryComment'.](https://app.codacy.com/manual/mundipagg/ecommerce-module-core/pullRequest?prid=4926694)